### PR TITLE
add error module (fixes #95)

### LINKED ISSUE #95: Add error module
I'm seeing errors from the smbus2 library and it would be nice to have our own uniform exceptions for that, in case smbus2 or python decides to change their particular errors at some point.

```
 File "/Users/foo/projects/bar/ssd1306/oled/device.py", line 66, in __init__
    self._serial_interface = serial_interface or i2c()
  File "/Users/foo/projects/bar/ssd1306/oled/serial.py", line 42, in __init__
    self._bus = bus or smbus2.SMBus(port)
  File "/Users/foo/.virtualenvs/bar/lib/python3.5/site-packages/smbus2/smbus2.py", line 132, in __init__
    self.open(bus)
  File "/Users/foo/.virtualenvs/bar/lib/python3.5/site-packages/smbus2/smbus2.py", line 142, in open
    self.fd = os.open("/dev/i2c-{}".format(bus), os.O_RDWR)
FileNotFoundError: [Errno 2] No such file or directory: '/dev/i2c-1'
```

### PR DESCRIPTION
**Pull request by thijstriemstra**

### CODE REVIEW

File: oled/error.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2016 Richard Hull and contributors
+# See LICENSE.rst for details.
```
Comment (thijstriemstra) [inline oled/error.py]: What do you think about adding a header like this instead of the big chunk of text? Other projects with the MIT license like Twisted do the same.

Comment (rm-hull) [inline oled/error.py]: Yep thats fine,

File: doc/api-documentation.rst [hunk 2]
```diff
@@ -53,4 +53,11 @@ API Documentation
 .. automodule:: oled.virtual
     :members:
     :undoc-members:
-    :show-inheritance:
\ No newline at end of file
+    :show-inheritance:
+
+:mod:`oled.error`
```
Comment (rm-hull) [inline doc/api-documentation.rst]: @thijstriemstra Can you reorder so that these sections are sorted alphabetically ?

File: oled/serial.py [hunk 3]
```diff
@@ -22,23 +22,37 @@
...
        2. If ``bus`` is provided, there is an implicit expectation
           that it has already been opened.
-          """
+    """
     def __init__(self, bus=None, port=1, address=0x3C):
         import smbus2
         self._cmd_mode = 0x00
         self._data_mode = 0x40
-        self._bus = bus or smbus2.SMBus(port)
+        try:
```
Comment (rm-hull) [inline oled/serial.py]: Should also remove the try/except IOError in the ssd1306 `__init__` method and handle here instead

Comment (rm-hull) [inline oled/serial.py]: same goes for `__init__` methods in sh1106 and ss1331 devices

File: oled/error.py (hunk 1 above)
Comment (rm-hull) [inline oled/error.py]: do it as a separate PR though please

File: oled/serial.py (hunk 3 above)
Comment (thijstriemstra) [inline oled/serial.py]: alright

File: doc/api-documentation.rst (hunk 2 above)
Comment (thijstriemstra) [inline doc/api-documentation.rst]: yes will do. Perhaps we should also limit the API to the parts that developers actually use, so hide the mixins, threading etc module?

File: oled/error.py (hunk 1 above)
Comment (thijstriemstra) [inline oled/error.py]: Opened #98 for that.
